indico/modules/events/registration/controllers/display.py

```diff
@@ -240,6 +253,43 @@ def _process(self):
         )
 
 
+class RHParticipantListPictureDownload(RHParticipantList):
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.registration
+        }
+    }
+
+    def _check_access(self):
+        RHParticipantList._check_access(self)
+        is_participant = self.registration.event.is_user_registered(session.user)
+        if not self.registration.is_publishable(is_participant):
+            raise Forbidden
+
+    def _process_args(self):
+        RHParticipantList._process_args(self)
+        regform_strategy = (joinedload('registration_form')
+                            .load_only('id', 'event_id', 'publish_registrations_duration',
+                                       'publish_registrations_public', 'publish_registrations_participants'))
+        data_strategy = selectinload('data').joinedload('field_data').joinedload('field')
+        self.registration = (Registration.query
```

Comment:
FYI I changed this to behave similar like `RHRegistrationAvatar`: Load only the columns we need, and avoid unnecessary queries. In case of large participant lists we may be fetching many of those images, so minimizing DB queries is really useful there.

Reply:
Done.